api/models/api.articles.models.js

```diff
@@ -2,12 +2,17 @@ const db = require("../../db/connection");
 
 exports.fetchArticlebyId = (article_id) => {
   return db
-    .query("SELECT * FROM articles WHERE article_id = $1", [article_id])
+    .query(`SELECT * FROM articles WHERE article_id = $1 `, [article_id])
     .then((article) => {
-      if (article.rows.length === 0)
+      if (article.rows.length === 0) {
         return Promise.reject({ status: 404, msg: "bad request" });
-      return article.rows[0];
-    });
+      }
+      const comments = db.query('SELECT * FROM comments WHERE article_id =$1',[article_id])
+      return Promise.all([article.rows[0], comments]) ;
+    }).then(([article, commentsArr])=>{
+      article.comment_count = commentsArr.rows.length
+      return article
+    })
 };
 
```

Comment:
To solve this, you need to use your SQL knowledge. 
You have articles, which connect to an article id in the comments table. 
You can make use of queries/JOINs to get this data - one of the advantages of having a relational database. 

This may solve/pass the tests, but doesn't show that you know how to do more complicated SQL statements, which would is one of the things you want in your portfolio piece.